### PR TITLE
[tools/model_explorer_circle] Introduce circle adapter for model expl…

### PR DESCRIPTION
**Pull request by batcheu**
This PR adds the initial version of circle adapter for model explorer.

---

issue: #14307 
draft: #14318

### CODE REVIEW

File: tools/model_explorer_circle/src/circle_adapter/main.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Model Explorer adapter for Circle models."""
+
+from typing import Dict
+from model_explorer import Adapter, AdapterMetadata, ModelExplorerGraphs, graph_builder
+
+
+class CicleAdapter(Adapter):
```
Comment (seockho-kim) [inline tools/model_explorer_circle/src/circle_adapter/main.py:22]: typo? CircleAdapter

File: tools/model_explorer_circle/pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+[project]
```
Comment (seanshpark) [inline tools/model_explorer_circle/pyproject.toml:1]: I think we can split this file to another PR

Comment (seanshpark) [reply tools/model_explorer_circle/pyproject.toml]: I think it is ok with `.gitignore` as initial file.
As I know, `pyproject.toml` is for publication so we can add this file for that purpose.

**Review by seanshpark — APPROVED** [review]
LGTM thank you!

**Review by seockho-kim — APPROVED** [review]
LGTM
